### PR TITLE
feat: allow user to configure "setAxeTypeRequired" to restrict "setAxeRequired" to require a specific axe type

### PR DESCRIPTION
**Pull request by matthamil**
FYI, I haven't compiled this code nor tested it myself. I will do this soon. I quickly threw this together so I could submit it as a PR to your plugin :). Feel free to edit this or leave feedback if you want me to make changes.

This introduces a `setAxeTypeRequired` option. When the `setAxeRequired` option is `true`, if the user sets a value between `1` and `6` for `setAxeTypeRequired`, the specific axe type associated with that number will be required for the plugin to chop down the whole tree. The `setAxeTypeRequired` is optional.

The motivation for this change is to prevent the need for users of this plugin to set up protections for areas like user-made structures that use logs. This lets players use any other axe type when editing their own structures, and use the specific axe type when chopping down trees.

See the `AxeTypeInput` in `src/objs/AxeType.java` for input options.

### CODE REVIEW

File: src/main/TreeCapitator.java [hunk 1]
```diff
@@ -481,10 +507,11 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 							accentColor + "Invincible replant: " + textColor
 									+ (invincibleReplant ? "enabled" : "disabled"),
 							accentColor + "Axe Needed: " + textColor + (axeNeeded ? "yes" : "no"),
+                            accentColor + "Axe Type Required: " + textColor + (AxeTypeInput.isValid(axeTypeRequired) ? axeTypeRequired : "disabled")
 							accentColor + "Axe Damaged: " + textColor + (axeNeeded ? "yes" : "no"),
 							accentColor + "Damage Axe: " + textColor + (damageAxe ? "yes" : "no"),
-							accentColor + "Break Axe: " + textColor + (breakAxe ? "yes" : "no"), 
-							accentColor + "Ignore Leaves: " + textColor + (ignoreLeaves ? "yes" : "no"), 
+							accentColor + "Break Axe: " + textColor + (breakAxe ? "yes" : "no"),
+							accentColor + "Ignore Leaves: " + textColor + (ignoreLeaves ? "yes" : "no"),
```
Comment (matthamil) [inline src/main/TreeCapitator.java:512]: No actual changes on L513 and L514. Looks like my editor removed some trailing whitespace.

File: src/main/TreeCapitator.java [hunk 2]
```diff
@@ -1104,7 +1166,7 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 
 	/**
 	 * Deals 1 damage to an item, if possible
-	 * 
+	 *
```
Comment (matthamil) [inline src/main/TreeCapitator.java:1167]: No change here either. Editor removed trailing whitespace.

**Comment by Cristichi** [comment]
Hi. I'm deeply sorry but the mechanic is highly unnecessary and contrary to the spirit of this plugin, which is boosting the early game to make it more quick. Also, while the implementation probably works, there is a lot of unnecessary code like the axetype class (remember we got the Material class, and I'd rather use the main TreeCapitator.java if I'd only need to implement 1 method to be honest).

I'll be closing this pull for now but thank you for your interest. Have a nice day.

**Comment by matthamil** [comment]
Fair enough. The axetype isnt unnecessary though. This enum-like static class frees us from using magic strings, like the existing checks for items to contain "_AXE" in their names.

**Comment by Cristichi** [comment]
> Fair enough. The axetype isnt unnecessary though. This enum-like static class frees us from using magic strings, like the existing checks for items to contain "_AXE" in their names.

As I understand it right now, magic strings are defined as a string the dev thinks will never come as an input but it probably will in the future, which I don't really think applies here. Either that or I'm the fool dev that thinks it probably won't, anyhow I wouldn't define it as a magic string more than any other string I'm using in the code tbh.

**Comment by matthamil** [comment]
I do want to follow up and say thank you for taking a look at this :). And thank you for maintaining this cool plugin!

I'll try to explain what I meant better. If any of this is old hat to you, I do apologize as I'm just sharing this as food for thought. I was specifically referring to the meaning of magic string that refers to using a literal value without assigning it to a variable. So instead of seeing if the item name contains an arbitrary string `"_AXE"`, you can use a reference that has semantic meaning, like `AxeType.STONE`. Currently (on `master`), using an enum like `AxeType` isn't necessary. However, for a feature like this to enable/disable the tree cutting functionality for a specific axe, it makes sense to check for specific axes. Because of this, I bundled up all of the axe names into an enum. This makes it easier than having `"WOODEN_AXE"`, `"STONE_AXE"`, etc. throughout the code in multiple places. The problem with introducing an additional class with static members like `AxeType` is that Java convention says to have one class per file, which is why I didn't put everything into `TreeCapitator.java`. You could set those as static values within `TreeCapitator`—there's no functional difference. Ultimately, you don't need an enum-like class, this is all depends on what you deem as readable/maintainable.

Anyway, I realized that I can tell my players to use `/tc toggle` to toggle the plugin. The motivation behind this change originally was so that players didn't accidentally mess up their own buildings while using an axe. This would let me tell players "Hey, if you are woodcutting, use this specific axe. If you are working on a building, use any other axe so you don't cut down your whole (or part of your) house." But in retrospect, it's really just easier to tell players to use the command.

**Pull request closed without merging**